### PR TITLE
Allow RTT & GDB to run concurrently!

### PR DESCRIPTION
**Pull request by Yatekii**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -464,29 +464,33 @@ fn main_try() -> Result<()> {
...
-    } else if config.rtt.enabled {
-        let session = Arc::new(Mutex::new(session));
+        let gdb_connection_string = config.gdb.gdb_connection_string.clone();
+        let session = session.clone();
+        std::thread::spawn(move || {
+            let gdb_connection_string = gdb_connection_string
+                .as_deref()
+                .or_else(|| Some("localhost:1337"));
+            // This next unwrap will always resolve as the connection string is always Some(T).
+            log::info!(
```
Comment (Tiwalun) [inline src/main.rs:483]: What's the reason for this change? Is there a threading issue with the `logging::println` call?

**Review by Tiwalun — CHANGES_REQUESTED** [review]
👍 Looking forward to RTT and GDB at the same time.

I only have some minor comments.

File: src/main.rs (hunk 1 above)
Comment (Yatekii) [reply src/main.rs:483]: Yes, this writes into the RTTUI, messing with the entire layout, rendering RTT unusable :/ I have not fond a way to swallow the logs from the GDB thread yet :/
Currently the GDB server also emits a lot of warnings that maybe should be debug logs. We sould clean this up.

File: src/main.rs [hunk 2]
```diff
@@ -466,29 +466,27 @@ fn main_try() -> Result<()> {
...
-            logging::eprintln("During the execution of GDB an error was encountered:");
-            logging::eprintln(format!("{:?}", e));
-        }
-    } else if config.rtt.enabled {
+        let gdb_connection_string = config.gdb.gdb_connection_string.clone();
+        let session = session.clone();
+        std::thread::spawn(move || {
+            let gdb_connection_string = gdb_connection_string
+                .as_deref()
+                .or_else(|| Some("127.0.0.1:1337"));
```
Comment (Tiwalun) [inline src/main.rs:477]:
```suggestion
                .or_else(Some("127.0.0.1:1337"));
```

**Review by Tiwalun — APPROVED** [review]
Looks good!

File: src/main.rs (hunk 2 above)
Comment (Tiwalun) [reply src/main.rs:477]: Just to fix the clippy lint.
